python/tchannel/tornado/connection.py

```diff
@@ -554,4 +574,19 @@ def send_request(self, request):
         future = tornado.gen.Future()
         self._outstanding[request.id] = future
         self.post_request(request)
+
+        def adapt_tracing(f):
+            # fetch the request tracing for response
+            f.result().tracing = request.tracing
```

Comment:
No. You're not using a future wrong if you have multiple consumers waiting on it. It's part of the design to have multiple consumers waiting on the result of an asynchronous operation. We do it all the time. You can set up a callback on a future to perform some cleanup in the calling context based on the result of the future and still return that same future up the stack where it's consumed as part of the main flow. 

And what I'm asking here is to not mutate the object at all. You shouldn't be mutating an object out of band in a callback that has no guarantee of ordering.